lightning/src/chain/keysinterface.rs

```diff
@@ -1094,6 +1131,11 @@ impl KeysInterface for KeysManager {
 	fn sign_invoice(&self, invoice_preimage: Vec<u8>) -> Result<RecoverableSignature, ()> {
 		Ok(self.secp_ctx.sign_recoverable(&hash_to_message!(&Sha256::hash(&invoice_preimage)), &self.get_node_secret()))
 	}
+
+	fn get_phantom_secret(&self, _scid: u64) -> Result<SecretKey, ()> {
+		let hkdf = hkdf_extract_expand(b"LDK Phantom Node Secret Key Expansion", &self.inbound_payment_key.0, 1);
```

Comment:
Is it the idea to later include the fake scid as a term in the generation of the phantom node key?

Reply:
Idea is that it's up to the implementer of `KeysInterface`. Adding the `scid` parameter adds the flexibility if they want to do it that way